### PR TITLE
Ai death bug fix

### PR DESCRIPTION
**Pull request by Tomerd-afk**
Fixed AI so that effects of death are reset when they next spawn. Also added minor name changes

### CODE REVIEW

File: Assets/Scripts/AI/Ai.cs [hunk 1]
```diff
@@ -275,6 +264,9 @@ public void NewLife()
     {
         alive = true;
         hp.Init(StartingHP);
+        animationStateController.SetAlive(true);
+        rb.detectCollisions = true;
+        rb.constraints = RigidbodyConstraints.None;
     }// this restets the enemies HP and sets them to alive;
```
Comment (Tomerd-afk) [inline Assets/Scripts/AI/Ai.cs:270]: This is the bulk of the fix
TODO: Add an interface for Animation state controllers in later PR

File: Assets/Scripts/AI/Ai.cs [hunk 2]
```diff
@@ -15,16 +15,15 @@ public void op_ProcessCompleted(SelfDespawn entity)
...
     }
 
-
-
     public GameObject target;
     public CyborgAnimationStateController animationStateController;
     public Rigidbody rb;
     public Gun myGun;
-    public Health hp;
+    public Health hp;
```
Comment (toryjleo) [inline Assets/Scripts/AI/Ai.cs:22]: remove trailing whitespace

File: Assets/Scripts/AI/Ai.cs [hunk 3]
```diff
@@ -15,16 +15,15 @@ public void op_ProcessCompleted(SelfDespawn entity)
...
     public GameObject target;
     public CyborgAnimationStateController animationStateController;
     public Rigidbody rb;
     public Gun myGun;
-    public Health hp;
+    public Health hp; 
     public List<Condition> activeConditions = new List<Condition>();
 
     public float StartingHP;
+    public float CurrentHP;
```
Comment (toryjleo) [inline Assets/Scripts/AI/Ai.cs:26]: This appears to be unused

**Review by toryjleo — CHANGES_REQUESTED** [review]

**Review by Tomerd-afk — COMMENTED** [review]
All set :shipit:

**Review by toryjleo — APPROVED** [review]

**Pull request merged**